oai-pmh-handler/src/main/java/no/sikt/nva/oai/pmh/handler/oaipmh/handlers/GetRecordRequestHandler.java

```diff
@@ -16,11 +19,17 @@
 public class GetRecordRequestHandler implements OaiPmhRequestHandler<GetRecordRequest> {
   private final ResourceRepository resourceRepository;
   private final RecordTransformer recordTransformer;
+  private final Pattern localIdPattern;
 
   public GetRecordRequestHandler(
-      ResourceRepository resourceRepository, RecordTransformer recordTransformer) {
+      ResourceRepository resourceRepository,
+      RecordTransformer recordTransformer,
+      URI identifierBaseUri) {
     this.resourceRepository = resourceRepository;
     this.recordTransformer = recordTransformer;
+    this.localIdPattern =
+        Pattern.compile(
+            "^" + identifierBaseUri.toString().replaceAll("\\.", "\\\\.") + "/([^/]+)$");
```

Comment:
Can you not just do UriWrapper::lastPathElement?

Reply:
We could, but we would act on every weird domain uri out there. I think we should make sure the identifier points at where we expect before doing a search.